File: location-service/internal/domain/location.go
package domain

import "time"

// Location represents a user's geographic position.
type Location struct {
	UserID    string    `json:"user_id"`
	Latitude  float64   `json:"latitude"`
	Longitude float64   `json:"longitude"`
	UpdatedAt time.Time `json:"updated_at"`
}

// MapPost represents a permanent map pin with user-generated content.
type MapPost struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	Title     string    `json:"title"`
	Content   string    `json:"content,omitempty"`
	Category  string    `json:"category"` // RESTAURANT, SOCIAL_LIFE, EVENT
	ImageURLs []string  `json:"image_urls,omitempty"`
	Latitude  float64   `json:"latitude"`
	Longitude float64   `json:"longitude"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: web/src/types/location.ts
// ─── Location / Map API Types ──────────────────────────────────────────

/** Payload for PUT /v1/location — updates a user's real-time position */
export interface LocationUpdate {
  user_id: string
  latitude: number
  longitude: number
  updated_at?: string
}

/** Entry in the frontend location log panel */
export interface LocationLogEntry {
  timestamp: string
  latitude: number
  longitude: number
  source: "click" | "drag" | "simulation" | "api" | "websocket"
}

/** A map post (pin) fetched from or sent to the Location Service API */
export interface MapPost {
  id: string
  user_id: string
  title: string
  content?: string
  category: "RESTAURANT" | "SOCIAL_LIFE" | "EVENT"
  image_urls?: string[]
  latitude: number
  longitude: number
  created_at: string
  updated_at: string
}

/** Category key for map posts */
export type PostCategory = "RESTAURANT" | "SOCIAL_LIFE" | "EVENT"

/** A simulated other user on the map */
export interface SimulatedUser {
  user_id: string
  name: string
  color: "#f59e0b" | "#10b981" | "#8b5cf6"
  lat: number
  lng: number
}

/** Payload for creating a new map post (POST /v1/posts) */
export interface CreatePostPayload {
  title: string
  content?: string
  category: "RESTAURANT" | "SOCIAL_LIFE" | "EVENT"
  image_urls?: string[]
  latitude: number
  longitude: number
}
